60008 bfa

Baschtl77 opened the pull request:
First version for BFA.
- Fixes API changes
- Added shaman as valid player class
- Shaman spells have no localization for korean and russian yet :/

coenvdwel commented on Rewatch.lua:
```diff
@@ -1632,7 +1822,7 @@ rewatch_events:SetScript("OnUpdate", function()
 				v["ManaBar"]:SetMinMaxValues(0, x); v["ManaBar"]:SetValue(y);
 				-- fade when out of range
-				if(IsSpellInRange(rewatch_loc["healingtouch"], v["Player"]) == 1) then
+				if(IsSpellInRange(rewatch_loc["regrowth"], v["Player"]) == 1 or IsSpellInRange(rewatch_loc["healingwave"], v["Player"]) == 1) then
```
We have some logic above from earlier which also checks for a non-cd-spell - best to store it locally and use here, as well.

coenvdwel commented:
Hi @Baschtl77! All your changes are good and well, don't be scared now that I close your PR. Please read my review summary (and the details if you want, optional) and check out new version 61000 branch!

By the way: I use GitHub for my work on a daily basis, for almost 10 years. I realize my review may appear blunt or unfriendly if you are not familiar with code- or peer reviews. Please know that I'm really happy with the changes, and the feedback is only meant to built knowledge together! 💪

Baschtl77 commented:
Hey Coen,

 

thanks a lot for your large review  - i didnt even dare to hope for a real review ^^'

Sorry, I didnt have time the past days to take a look at the new 61000 branch. (Enjoyed the great weather with my wife ^^')

Just saw that you allready did all the fixes from your review! Great!

 

Now i was able to test the 61000-version. There where some small syntax-errors, which i fixed. 

I also recognized that the shaman-decurse spell button was

not blured when no debuff is active - so i fixed that also.

 

I started a new pullrequest for 61000

 

Btw. I am using eclipse + lua-plugin.

The Source-format does not work, but it has syntax validation, which is often helpful.

I am not very used to lua, for that i dont know a common way for debugging or unittests :(

 

 

I wish you a nice sunday and again thanks a lot for your commitment at one of the greatest Addons :)

 

Sebastian

 

 

Gesendet: Donnerstag, 26. Juli 2018 um 23:32 Uhr
Von: "Coen van der Wel" <notifications@github.com>
An: coenvdwel/Rewatch <Rewatch@noreply.github.com>
Cc: Baschtl77 <github@lorr-and-order.com>, Author <author@noreply.github.com>
Betreff: Re: [coenvdwel/Rewatch] 60008 bfa (#5)


@coenvdwel requested changes on this pull request.

I cannot thank you enough for the effort to get this working! Great to see a community still active around my little AddOn.

I have taken all your changes (3 commits) and merged them into version 61000! 🎁

Then I made some additional changes, based on the feedback I give in this PR review.

Only two things remain:


	You left a bit of "todo" code: what's that?
	I... Could not even test if the syntax is OK, no account to tets it with xD


Please create a tracking branch off of my 61000 branch and see if everything works out for you! I'll await your final confirmation before I push to Curse!


In Locale/localization.esES.lua:

> @@ -123,4 +123,11 @@ if (GetLocale() == "esES") then
 	rewatch_loc["direBearForm"] = "Forma de oso temible";
 	rewatch_loc["catForm"] = "Forma felina";
 	
+	-- shaman spell names
+  rewatch_loc["riptide"] = "Mareas vivas";
+  rewatch_loc["purifyspirit"] = "Purificar esp�ritu";
+  rewatch_loc["healingsurge"] = "Oleada de sanaci\195\179n";


Well done, I usually forget this encoding xD


In Rewatch.lua:

> @@ -7,9 +7,14 @@
 -- Please give full credit when you want to redistribute or modify this addon!
 
 
-local rewatch_versioni = 60007;
+local rewatch_versioni = 60008;


Give yourself some credit! Version 6100, here we come! Shaman support is a major feature 🎉


In Rewatch.lua:

>  --------------------------------------------------------------------------------------------------------------[ FUNCTIONS ]----------------------
 
+-- create some more output to chatwindow
+-- only for debugging purpose
+local debug_enabled=false;


I'm a bit weird, usually need this indeed when debugging, but I always remove it from code after I fixed something... Not that my code (it's grown huge!) gets a pretty-award, but still...


In Rewatch.lua:

> @@ -89,16 +106,25 @@ function rewatch_OnLoad()
 				rewatch_load["OORAlpha"] = 0.2;
 			end;
 			if(rewatch_version < 60005) then
-				rewatch_load["ButtonSpells"] = { rewatch_loc["swiftmend"], rewatch_loc["naturescure"], rewatch_loc["ironbark"], rewatch_loc["healingtouch"], rewatch_loc["mushroom"] };
+				rewatch_load["ButtonSpells"] = { rewatch_loc["swiftmend"], rewatch_loc["naturescure"], rewatch_loc["ironbark"], rewatch_loc["mushroom"] };


No need to update for previous versions, they get overwritten.


In Rewatch.lua:

>  			
 			-- get spec properties
 			rewatch_loadInt["InRestoSpec"] = false;
-			if(GetSpecialization() == 4) then
+			if((GetSpecialization() == 4 and  select(3, UnitClass("player")) == 11) or (GetSpecialization() == 3 and  select(3, UnitClass("player")) == 7)) then


Cool! I moved these UnitClass calls to rewatch_loadInt variables though, I have a feeling that's better for performance.


In Rewatch.lua:

> @@ -143,10 +170,18 @@ function rewatch_OnLoad()
 			rewatch_loadInt["ShowIncomingHeals"] = rewatch_load["ShowIncomingHeals"];
 			rewatch_loadInt["ShowSelfFirst"] = rewatch_load["ShowSelfFirst"];
 			rewatch_loadInt["ButtonSpells"] = rewatch_load["ButtonSpells"];
+			-- -- special buttons  for current class: shaman (7) , druid (11)
+			rewatch_loadInt["ButtonSpells"..select(3, UnitClass("player"))] = rewatch_load["ButtonSpells"..select(3, UnitClass("player"))];


This... Is overwritten by either 175 or 176 😋


In Rewatch.lua:

>  			rewatch_loadInt["FrameColumns"] = rewatch_load["FrameColumns"];
 			rewatch_loadInt["LockP"] = true;
 			-- update later
 			rewatch_changed = true;
+			
+			rewatch_Debug_Message("playerid: "..select(3, UnitClass("player")).." has following buttons: "..table.concat(rewatch_loadInt["ButtonSpells"..select(3, UnitClass("player"))])); --DEBUG REMOVE ME


DEBUG REMOVE ME 😁


In Rewatch.lua:

> @@ -263,11 +303,30 @@ end;
 function rewatch_UpdateOffset()
 	if(rewatch_loadInt["Layout"] == "horizontal") then
 		rewatch_loadInt["FrameWidth"] = (rewatch_loadInt["SpellBarWidth"]) * (rewatch_loadInt["Scaling"]/100);
-		rewatch_loadInt["ButtonSize"] = (rewatch_loadInt["SpellBarWidth"] / table.getn(rewatch_loadInt["ButtonSpells"])) * (rewatch_loadInt["Scaling"]/100);
-		rewatch_loadInt["FrameHeight"] = ((rewatch_loadInt["SpellBarHeight"]*(3+rewatch_loadInt["WildGrowth"])) + rewatch_loadInt["HealthBarHeight"]) * (rewatch_loadInt["Scaling"]/100) + (rewatch_loadInt["ButtonSize"]*rewatch_loadInt["ShowButtons"]);
+		rewatch_loadInt["ButtonSize"] = (rewatch_loadInt["SpellBarWidth"] / table.getn(rewatch_loadInt["ButtonSpells"..select(3, UnitClass("player"))])) * (rewatch_loadInt["Scaling"]/100);
+		
+		-- druid
+		if( select(3, UnitClass("player")) == 11) then 


Ah, are you using Visual Studio Code for this? I'm just on good-ol' Notepad (ok, ++ version) and use tabs, instead of spaces.


In Rewatch.lua:

> @@ -427,7 +486,13 @@ end;
 -- return: void
 function rewatch_TriggerCooldown()
 	-- get global cooldown, and trigger it on all frames
-	local start, duration, enabled = GetSpellCooldown(rewatch_loc["healingtouch"]); -- some non-cd spell
+	local non_cd_spell = rewatch_loc["regrowth"];


Good solution! But on line 495 you always use Regrowth 🤣


In Rewatch.lua:

> -	for dy=0, my, -1 do
-		for dx=0, mx, 1 do
-			found, x, y = false, rewatch_loadInt["FrameWidth"]*dx, rewatch_loadInt["FrameHeight"]*dy;
-			-- check if there's a frame here already
-			for i, child in ipairs(children) do
-				if((child:GetLeft() and (i>1))) then
-					if((math.abs(x - (child:GetLeft()-frame:GetLeft())) < 1) and (math.abs(y - (child:GetTop()-frame:GetTop())) < 1)) then
-						found = true; break; --[[ break for children loop ]] end;
-				end;
-			end;
-			-- if not, we found a spot and we should break!
-			if(not found) then break; --[[ break for dxloop ]] end;
-		end;
-		if(not found) then break; --[[ break for dy loop ]] end;
-	end;
+	if (rewatch_loadInt["FrameWidth"] ~= nil and rewatch_loadInt["FrameHeight"] ~= nil) then


This is probably my worst piece of code in terms of maintainability! But eh, is this check necessary? Are those variables not always filled?


In Rewatch.lua:

> +      rewatch_Debug_Message("Got buff duration" .. select(6, UnitBuff(player, counter, filter)));
+
+      return select(6, UnitBuff(player, counter, filter));
+    end;
+  end;
+
+end;
+
+
+-- check if debuff is decursible 
+-- replaces 
+-- _, _, _, _, debuffType = UnitDebuff(playerId, spell);
+-- if((debuffType == "Curse") or (debuffType == "Poison") or (debuffType == "Magic" and rewatch_loadInt["InRestoSpec"])) then
+-- 
+
+--todo - make this work


This doesn't work?


In Rewatch.lua:

> @@ -840,6 +958,8 @@ end;
 function rewatch_AddPlayer(player, pet)
 	-- return if in combat or if the max amount of players is passed
 	if(rewatch_inCombat or ((rewatch_loadInt["MaxPlayers"] > 0) and (rewatch_loadInt["MaxPlayers"] < rewatch_f:GetNumChildren()))) then return -1; end;
+	--bakka: why not rearrange in combat ?


This is blocked by WOW API (at least, it used to be?), else you could create a big button in the middle of the screen saying "DO BEST SPELL" and you could program it to be the best in your DPS rotation, or whatever - they didn't want that 😄


In Rewatch.lua:

> @@ -890,7 +1010,9 @@ function rewatch_AddPlayer(player, pet)
 	
 	-- determine class
 	local classID, class, classColors;
-	if(UnitName("player") == player) then classID = 11; else classID = select(3, UnitClass(player)); end;
+	-- if(UnitName("player") == player) then classID = 11; else classID = select(3, UnitClass(player)); end;
+	-- player can bne shaman as well
+  classID = select(3, UnitClass(player));


I think I remember making this so 'weird' (instead of just always doing UnitClass) because sometimes it would give nil and you would end up with a black healthbar of your own.


In Rewatch.lua:

> @@ -1632,7 +1822,7 @@ rewatch_events:SetScript("OnUpdate", function()
 				x, y = UnitPowerMax(v["Player"]), UnitPower(v["Player"]);
 				v["ManaBar"]:SetMinMaxValues(0, x); v["ManaBar"]:SetValue(y);
 				-- fade when out of range
-				if(IsSpellInRange(rewatch_loc["healingtouch"], v["Player"]) == 1) then
+				if(IsSpellInRange(rewatch_loc["regrowth"], v["Player"]) == 1 or IsSpellInRange(rewatch_loc["healingwave"], v["Player"]) == 1) then


We have some logic above from earlier which also checks for a non-cd-spell - best to store it locally and use here, as well.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or mute the thread.
